softioc/device.py

```diff
@@ -1,19 +1,36 @@
+from inspect import iscoroutinefunction
 import os
 import time
 import ctypes
 from ctypes import *
 import numpy
+from threading import Event
 
 from . import alarm
 from . import fields
-from .imports import dbLoadDatabase, recGblResetAlarms, db_put_field
+from .imports import (
+    create_callback_capsule,
+    dbLoadDatabase,
+    signal_processing_complete,
+    recGblResetAlarms,
+    db_put_field,
+)
 from .device_core import DeviceSupportCore, RecordLookup
 
 
 # This is set from softioc.iocInit
 # dispatcher(func, *args) will queue a callback to happen
 dispatcher = None
 
+# Global blocking flag, used to mark asynchronous (False) or synchronous (True)
+# processing modes for Out records.
+# Default False to maintain behaviour from previous versions.
+blocking = False
+
+def SetBlocking(val):
+    global blocking
+    blocking = val
```

Comment:
It wouldn't hurt to return the old value of `blocking`.  This would then allow a coding style of the form:
```python
old_blocking = SetBlocking(new_blocking)
create_records()
SetBlocking(old_blocking)
```
which could even be wrapped up as a `with` using a context manager.

Not suggesting we write the context manager, just that we return the old state!

Reply:
Added in a0bce0ce330bb96274d4bbebff179c872165d7e6